### PR TITLE
Updating button documentation

### PR DESCRIPTION
**Pull request by heatherlarsen**
### :pushpin: Summary

If merged, this PR introduces adjustments to the `button` documentation.

**Preview link:** https://hds-website-git-heatherlarsen-component-button-hashicorp.vercel.app/components/button

### :hammer_and_wrench: Detailed description

Changes include:
- Added keywords
- Removing unused images
- Grammar and language fixes
- Aligning the button vs link content to the Helios voice and tone

### :link: External links

Jira ticket: [HDS-1351](https://hashicorp.atlassian.net/browse/HDS-1351)

***

### 👀 Reviewer's checklist:

- [ ] +1 Percy if applicable
- [ ] Confirm that PR has a changelog update via [Changesets](https://github.com/changesets/changesets) if needed

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

[HDS-1351]: https://hashicorp.atlassian.net/browse/HDS-1351?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: website/docs/components/button/partials/guidelines/guidelines.md [hunk 1]
```diff
@@ -6,109 +6,94 @@
...
+</Hds::ButtonSet>
 
-Button Sets are patterns when multiple buttons need to be displayed in a single row; not to be confused with Button Groups.
+## Button set
 
-  <Hds::ButtonSet>
-    <Hds::Button @color="primary" @text="Submit" />
-    <Hds::Button @color="secondary" @text="Cancel" />
-  </Hds::ButtonSet>
+ButtonSets are patterns when multiple Buttons need to be displayed in a single row.
```
Comment (alex-ju) [inline website/docs/components/button/partials/guidelines/guidelines.md:89]: Shall we link to the ButtonSet component here?

**Review by alex-ju — APPROVED** [review]
Looks great! Left only one suggestion

File: website/docs/components/button/partials/code/how-to-use.md [hunk 2]
```diff
@@ -1,45 +1,54 @@
...
+The basic invocation requires text to be passed:
 
 ```handlebars
 <Hds::Button @text="Copy to clipboard" />
 ```
 
 ### Add an icon
 
-To add an icon to your button, give the `@icon` any [icon](/foundations/icons/) name:
+To add an icon to your button, give the `@icon` any [Helios icon](/foundations/icons/library) name:
```
Comment (jorytindall) [inline website/docs/components/button/partials/code/how-to-use.md:13]:
```suggestion
To add an icon to your button, give the `@icon` any [Helios icon](/icons/library) name:
```
This is the correct link

**Review by jorytindall — APPROVED** [review]
Small comment on correcting the link, I'm hoping I didn't miss any others though

**Pull request merged**